packages/flame_tiled/lib/src/renderable_tile_map.dart

```diff
@@ -192,17 +269,161 @@ class RenderableTiledMap {
-  /// of this map. If no such layer is found, null is returned.
-  T? getLayer<T extends Layer>(String name) {
-    return map.layers
-        .firstWhereOrNull((layer) => layer is T && layer.name == name) as T?;
+  static Future<_RenderableLayer> load(
+    TileLayer layer,
+    TiledMap map,
+    Vector2 destTileSize,
+  ) async {
+    return _RenderableTileLayer(
+      layer,
+      map,
+      destTileSize,
+      await _loadImages(map),
+    );
+  }
+
+  static Future<Map<String, SpriteBatch>> _loadImages(TiledMap map) async {
+    final result = <String, SpriteBatch>{};
+
+    await Future.forEach(map.tiledImages(), (TiledImage img) async {
+      final src = img.source;
+      if (src != null) {
+        result[src] = await SpriteBatch.load(src);
+      }
+    });
+
+    return result;
+  }
+}
+
+class _RenderableImageLayer extends _RenderableLayer<ImageLayer> {
+  final Image _image;
+  late final ImageRepeat _repeat;
+  Rect _paintArea = Rect.zero;
+
+  _RenderableImageLayer(super.layer, this._image) {
+    _initImageRepeat();
+  }
+
+  @override
+  void handleResize(Vector2 canvasSize) {
+    _paintArea = Rect.fromLTWH(0, 0, canvasSize.x, canvasSize.y);
+  }
+
+  @override
+  void render(Canvas canvas, Camera? camera) {
+    canvas.save();
+
+    // this seems to match how the Tiled editor initially offsets image layers
+    canvas.translate(-_image.width + layer.offsetX, layer.offsetY);
+
+    if (camera != null) {
+      _applyParallaxOffset(canvas, camera, layer);
+    }
+
+    paintImage(
+      canvas: canvas,
+      rect: _paintArea,
+      image: _image,
+      opacity: layer.opacity,
+      alignment: Alignment.topLeft,
+      repeat: _repeat,
+    );
+
+    canvas.restore();
+  }
+
+  void _initImageRepeat() {
+    if (layer.repeatX && layer.repeatY) {
+      _repeat = ImageRepeat.repeat;
+    } else if (layer.repeatX) {
+      _repeat = ImageRepeat.repeatX;
+    } else if (layer.repeatY) {
+      _repeat = ImageRepeat.repeatY;
+    } else {
+      _repeat = ImageRepeat.noRepeat;
+    }
+  }
+
+  static Future<_RenderableLayer> load(
+    ImageLayer layer,
+    Camera? camera,
+  ) async {
+    return _RenderableImageLayer(
+      layer,
+      await Flame.images.load(layer.image.source!),
+    );
+  }
+}
+
+class _UnrenderableLayer extends _RenderableLayer {
+  _UnrenderableLayer(super.layer);
+
+  @override
+  void render(Canvas canvas, Camera? camera) {
+    // nothing to do
+  }
+
+  // ignore unrenderable layers when looping over the layers to render
+  @override
+  bool get visible => false;
+
+  static Future<_RenderableLayer> load(Layer layer) async {
+    return _UnrenderableLayer(layer);
+  }
+}
+
+/// Calculates the offset we need to apply to the canvas to compensate for
+/// parallax positioning and scroll for the layer and the current camera
+/// position
+/// https://doc.mapeditor.org/en/latest/manual/layers/#parallax-scrolling-factor
+void _applyParallaxOffset(Canvas canvas, Camera camera, Layer layer) {
```

Comment:
This is only used by one class right? I think it could be moved in to that class then.

Reply:
It's used in two different layer classes, but I can put it in the `_RenderableLayer` base class 👍

Reply:
Done